File: scripts/tokenize.py
from subprocess import Popen, PIPE, STDOUT
import sys
from os import listdir
from os.path import isfile, join
import fnmatch

def load_z_file(file_path):
    cmd = 'gzip -d -c %s' % (file_path)
    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
    return p.stdout.read()

def load_corpus(corpus_path):
    file_names = listdir(corpus_path)
    return map(lambda n: load_z_file("%s/%s" % (corpus_path, n)),
        filter(lambda n: fnmatch.fnmatch(n, '*.z'), file_names))

print(load_corpus("./corpus"))

File: scripts/tokenize.js
"use strict";

var co = require("co");
var bluebird = require("bluebird");
var fs = require("fs");
var path = require("path");
var childProcess = require('child_process');
bluebird.promisifyAll(fs);

const CORPUS_PATH = path.resolve(__dirname, "../corpus");

co(function*() {
  var corpusFiles = yield fs.readdirAsync(CORPUS_PATH);
  var documents = yield bluebird.map(corpusFiles.filter(fn => /\.z$/.test(fn)),
    fn => new Promise((resolve, reject) => {
      const cmd = 'bash -c "zcat < ' + path.join(CORPUS_PATH, fn) + '"';
      childProcess.exec(cmd, {maxBuffer: 1024 * 10240}, (err, stdout, stderr) => {
        if(err) {
          console.log(err)
          return reject(err);
        }
        resolve(stdout);
      })
    }), {concurrency: 5});
})
.catch(e => console.error(e));
